sdk/core/Azure.Core.TestFramework/src/TestActivitySourceListener.cs

```diff
@@ -16,11 +17,11 @@ public class TestActivitySourceListener: IDisposable
         public Queue<Activity> Activities { get; } =
             new Queue<Activity>();
 
-        public TestActivitySourceListener(string name) : this(source => source.Name == name)
+        public TestActivitySourceListener(string name, Action<Activity> activityStartedCallback = default) : this(source => source.Name.StartsWith(name), activityStartedCallback)
```

Comment:
StartsWith aligns better with the usage for activity source where wildcards are used so updating the test class, e.g.:
```c#
.AddSource("Azure.*") // Collect all traces from Azure SDK
```